File: 백준/Silver/1181. 단어 정렬/단어 정렬.py
import sys

# 입력을 받습니다.
input = sys.stdin.read
words = input().splitlines()[1:]  # 첫 번째 줄은 단어 개수이므로 제외

# 중복 제거를 위해 set 사용
words = list(set(words))

# 길이와 사전 순으로 정렬
words.sort(key=lambda x: (len(x), x))

# 결과 출력
for word in words:
    print(word)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : 'input.txt').toString().trim().split('\n');

const wordLength = input[0];
const words = input.slice(1);

words.sort((a,b) => {
    if(a.length === b.length) return a.localeCompare(b); // 문자열 정렬은 sort보단 localeCompare 메서드 사용
    return a.length - b.length;
})

const wordsList = new Set(words);

console.log([...wordsList].join('\n'))